src/lib/components/globalhealthbar/utils.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { RectRadius } from 'recharts/types/shape/Rectangle';
+import {
+  DATE_FORMATER,
+  TIME_FORMATER,
+  TIME_SECOND_FORMATER,
+} from '../date/FormattedDateTime';
+
+const oneHour = 60 * 60 * 1000;
+const oneDay = 24 * oneHour;
+
+export const setHistoryTooltipPosition = (
+  startDate: number,
+  endDate: number,
+  selectedDate: number | undefined,
+): string | null => {
+  if (selectedDate) {
+    const width = ((selectedDate - startDate) / (endDate - startDate)) * 600;
+    const leftPosition = width - 132 / 2;
+    return `auto auto -4px ${leftPosition}px`;
```

Comment:
From a reader perspective it looks like math black magic here, maybe we can introduce variables to explain what is
- 600
- 132
- -4px ?